packages/salesforce-adapter/src/client/client.ts

```diff
@@ -662,20 +665,24 @@ export default class SalesforceClient {
   /**
    * Updates salesforce metadata with the Deploy API
    * @param zip The package zip
+   * @param deployOptions Required salesforce deployment options
    * @returns The save result of the requested update
    */
   @throttle<ClientRateLimitConfig>({ bucketName: 'deploy' })
   @logDecorator()
   @requiresLogin()
-  public async deploy(zip: Buffer): Promise<DeployResult> {
+  public async deploy(zip: Buffer, deployOptions: DeployOptions): Promise<DeployResult> {
     this.setDeployPollingTimeout()
-    const defaultDeployOptions = { rollbackOnError: true, ignoreWarnings: true }
-    const optionsToSend = ['rollbackOnError', 'ignoreWarnings', 'purgeOnDelete',
-      'checkOnly', 'testLevel', 'runTests']
+    const configOverrides = { rollbackOnError: true, ignoreWarnings: true }
+    const { checkOnly = false } = deployOptions
+    const optionsToSend = ['rollbackOnError', 'ignoreWarnings', 'purgeOnDelete', 'testLevel', 'runTests']
     const deployResult = flatValues(
       await this.conn.metadata.deploy(
         zip,
-        _.merge(defaultDeployOptions, _.pick(this.config?.deploy, optionsToSend)),
+        {
+          ..._.merge(configOverrides, _.pick(this.config?.deploy, optionsToSend)),
+          checkOnly,
+        },
```

Comment:
why mix `_.merge` and `{ ... }`?
```suggestion
       _.merge(configOverrides, _.pick(this.config?.deploy, optionsToSend), { checkOnly })
```
or
```suggestion
        {
          ...configOverrides,
          ..._.pick(this.config?.deploy, optionsToSend),
          checkOnly,
        },
```

Reply:
v